src/doc/rustdoc/src/documentation-tests.md

```diff
@@ -233,6 +246,16 @@ not actually pass as a test.
 # fn foo() {}
 ```
 
+`compile_fail` tells `rustdoc` that the compilation should fail. If it
+compiles, then the test will fail.
```

Comment:
Still looking for a note here about how things may start compiling in later releases.